rueidiscompat/command.go

```diff
@@ -2109,6 +2112,134 @@ func (cmd *ClusterSlotsCmd) Result() ([]ClusterSlot, error) {
 	return cmd.val, cmd.err
 }
 
+func newClusterShardsCmd(res rueidis.RedisResult) *ClusterShardsCmd {
+	arr, err := res.ToArray()
+	if err != nil {
+		return &ClusterShardsCmd{err: err}
+	}
+	val := make([]ClusterShard, 0, len(arr))
+	for _, v := range arr {
+		dict, err := v.ToMap()
+		if err != nil {
+			return &ClusterShardsCmd{err: err}
+		}
+		var (
+			start, end int64
+			nodesArr   []ClusterNode
+		)
+		{
+			slots, ok := dict["slots"]
+			if !ok {
+				return &ClusterShardsCmd{err: errors.New("slots not found")}
+			}
+			arr, err := slots.ToArray()
+			if err != nil {
+				return &ClusterShardsCmd{err: err}
+			}
+			if len(arr) != 2 {
+				return &ClusterShardsCmd{err: fmt.Errorf("got %d, expected 2", len(arr))}
+			}
+			start, err = arr[0].AsInt64()
+			if err != nil {
+				return &ClusterShardsCmd{err: err}
+			}
+			end, err = arr[1].AsInt64()
+			if err != nil {
+				return &ClusterShardsCmd{err: err}
+			}
+		}
+		{
+			nodes, ok := dict["nodes"]
+			if !ok {
+				return &ClusterShardsCmd{err: errors.New("nodes not found")}
+			}
+			arr, err := nodes.ToArray()
+			if err != nil {
+				return &ClusterShardsCmd{err: err}
+			}
+			nodesArr = make([]ClusterNode, len(arr))
+			for i := 0; i < len(arr); i++ {
+				var (
+					err  error
+					elem rueidis.RedisMessage
+				)
+				nodeMap, err := arr[i].ToMap()
+				if err != nil {
+					return &ClusterShardsCmd{err: err}
+				}
+
+				elem = nodeMap["id"]
+				nodesArr[i].ID, err = elem.ToString()
+				if err != nil {
+					return &ClusterShardsCmd{err: err}
+				}
+
+				elem = nodeMap["ip"]
+				ip, err := elem.ToString()
+				if err != nil {
+					return &ClusterShardsCmd{err: err}
+				}
+				nodesArr[i].Host = ip
+
+				elem, ok = nodeMap["port"]
+				if ok {
+					port, err := elem.AsInt64()
+					if err != nil {
+						return &ClusterShardsCmd{err: err}
+					}
+					nodesArr[i].Port = port
+				}
+
+				elem, ok = nodeMap["tls-port"]
+				if ok {
+					port, err := elem.AsInt64()
+					if err != nil {
+						return &ClusterShardsCmd{err: err}
+					}
+					nodesArr[i].TlsPort = port
+				}
+			}
+		}
+		val = append(val, ClusterShard{
+			Start: start,
+			End:   end,
+			Nodes: nodesArr,
+		})
+	}
+	return &ClusterShardsCmd{val: val, err: err}
+}
+
+type ClusterShard struct {
```

Comment:
This struct should be aligned with go-redis

```suggestion
type SlotRange struct {
	Start int64
	End   int64
}

type Node struct {
	ID                string
	Endpoint          string
	IP                string
	Hostname          string
	Port              int64
	TLSPort           int64
	Role              string
	ReplicationOffset int64
	Health            string
}

type ClusterShard struct {
	Slots []SlotRange
	Nodes []Node
}
```